### DISCUSSION TITLE
Raw file language analysis doesn't load gitattributes data properly

### DISCUSSION THREAD

**Discussion by Nixinova** [lowlighter/metrics#898] [Q&A] [answered]
With the recent change to using `fileContent` instead of cloning the repo, gitattributes data is no longer parsed, so vendored files are now affecting my language results. For example [this file](https://github.com/github/linguist/blob/7a2a2ebfc02d98e3748b7f4997c86173f87eb2f9/samples/Beef/Program.bf) which is marked vendored has put 6% Beef into my stats. We need to think of a way to incorporate gitattributes data into this language analysis.

**Comment by Nixinova** [comment, marked as answer]
Maybe I need to also have an `attributesContent` option?

**Reply by lowlighter** [reply]
Yeah I think it may be the most suitable solution 👍 

However the recent change should only impact the code snippet plugin I think, the languages plugin hasn't been updated yet (it stills clones repositories for both recent/most used stats, along with the `.gitattributes`)

**Reply by Nixinova** [reply]
Oh, does it? Hm. Then I have no idea where Beef is coming from here. I assume this problem is on my end then.
<img width="379" alt="image" src="https://user-images.githubusercontent.com/42429413/154571390-eb49e7d9-d072-471c-b164-f5a488fbda3b.png">

**Reply by Nixinova** [reply]
Found the issue maybe. But not classifying `samples/` as vendored seems to be a bug that has been in linguist-js for ages, no idea why it only showed up now...

**Reply by Nixinova** [reply]
Well that was a rabbit hole https://github.com/Nixinova/Linguist/commit/161a40a3be2d9c833e0a811e47d6209c285f5317. Should be fixed in 2.3.2

**Reply by lowlighter** [reply]
Nice, thanks a lot for looking into it 😄 !
